module/VuFind/src/VuFind/Recommend/ConsortialVuFindFactory.php

```diff
@@ -70,9 +70,12 @@ public function __invoke(
             throw new \Exception('Unexpected options passed to factory.');
         }
 
+        $externalVuFind = $container->get(\VuFind\Connection\ExternalVuFind::class);
+        $cachingDownloader = $container->get(\VuFind\Http\CachingDownloader::class);
```

Comment:
I think it makes more sense to set this in the factory used by ExternalVuFind instead of doing it here (it's weird for one service's factory to modify a different service as a side effect), and you'll _have to_ if you take my advice regarding the constructor.

Reply:
Agreed, done.